internal/commands/exec/command.go

```diff
@@ -165,6 +170,37 @@ func (c *Command) Run(args []string) (ret int) {
 		bearerToken = strings.TrimSpace(string(data))
 	}
 
+	addresses, grpcPort, err := c.parseDiscoveryAddresses()
+	if errors.Is(err, discoverySyntaxError) {
```

Comment:
Unless you're seeing issues with passing ips/dns strings to the discovery library, again I don't think we need to do the `exec=` check.